charts/argo-cd/values.yaml

```diff
@@ -259,6 +259,19 @@ dex:
   ##
   podLabels: {}
 
+  ## Probes for Dex server
+  ## Supported from Dex >= 2.28.0
+  probes:
+    enabled: false
+    livenessProbe:
+      httpGet:
+        path: /healthz/live
+        port: 5558
```

Comment:
@mkilchhofer This kinda feels like a code smell to me. As a user I need to know which container port exposes the probe. I have similar feel towards `containerPortHttp` and other values because Dex is initialized via `argocd-dex` utility that hardcodes ports. Is there any real use case when user should be able to redefine ports for this deployment? Another subjective code smell is hiding the metrics container port based on configuration. This port still exists and is used by the container even when you have `dex.metrics.enabled: false`

Reply:
If I can propose another change for different PR that relates to metrics port and how `metrics.enable` and `serviceMonitor.enable` currently behaves I would go with Helm 3 API capabilities check that autodetects installed Prometheus and kept metrics port on deployment always enabled / used with alias name instead of port number. What do you think?

```yaml
{{- if and .metrics.enabled .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" }}
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
{{- end }}
```

Reply:
Hmm. Using `.Capabilities` in context of ArgoCD is not a good idea. We had several issues, where users deployed the argo-cd chart through another ArgoCD instance. ArgoCD only uses the `helm template` subcommand which does not contact the real kubernetes API. Therefore the `.Capabilities` dict isn't populated with values reflecting the kubernetes cluster.

Reply:
What about removing the conditional `{{- if .Values.dex.metrics.enabled }}` in the deployment.yaml and reference the  probe like this:
```yaml
      httpGet:
        path: /healthz/live
        port: metrics
```
?

Reply:
Sure we can reference it with name instead of port number. What about the ports for Dex? Do you want to keep them configurable via values.yaml even when they cannot be changed at all?

Regarding API capabilities - we are using this approach on our production servers to detect Prometheus installations automatically with Argo CD 2.0.5 and it works fine. Maybe this was problem on some older version and now Argo invokes `helm template` with  `--api-versions` switch.  I'll try to take a look into this when I have a moment.

Reply:
It's supported - see here: https://github.com/argoproj/argo-cd/blob/9476ab5e18be34a8cfc81150c94d08eefe3cee25/util/helm/cmd.go#L321